### PR TITLE
DOC restructure contributing environment guide

### LINKED ISSUE #50134: BUILD: No module named 'pandas._libs.interval' after building container from .devcontainer.json
I cloned the repo and used the `.devcontainer.json` with VS Code to build a dev docker, on my Linux laptop.    
It seemed to run well until I tried `import pandas` and got `ModuleNotFoundError: No module named 'pandas._libs.interval'`.  

I tried the commands recommended here: https://stackoverflow.com/a/68935982
It didnt work.  

The problem is easy to reproduce, because it also happens in Github Codespace, which, I suppose, also use  `.devcontainer.json`.

### Installation check

- [X] I have read the [installation guide](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#installing-pandas).

### Platform

Linux-5.19.16-76051916-generic-x86_64-with-glibc2.31

### Installation Method

Built from source

### pandas Version

building from latest .devcontainer.json on main

### Python Version

3.10.8

### Installation Logs

<details>

Replace this line with the installation logs.

</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #50134 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Hey @noatamir , thoughts on this? I think it's clearer to split it up into 3 steps:

image

I've also removed some parts which are arguably unnecessary

### CODE REVIEW

File: doc/source/development/contributing_environment.rst [hunk 1]
```diff
@@ -246,3 +187,25 @@ See https://code.visualstudio.com/docs/remote/containers for details.
...
+   $ python
+   >>> import pandas
+   >>> print(pandas.__version__)  # note: the exact output may differ
+   2.0.0.dev0+880.g2b9e661fbb.dirty
+
+This will create the new environment, and not touch any of your existing environments,
+nor any existing Python installation.
+
+Note that you will need to repeat this step each time the C extensions change, for example
+if you modified them or if you did a fetch and merge from ``upstream/main``.
```
Comment (mroeschke) [inline doc/source/development/contributing_environment.rst:211]:
```suggestion
if you modified any file in `pandas/_libs` or if you did a fetch and merge from ``upstream/main``.
```

File: doc/source/development/contributing_environment.rst [hunk 2]
```diff
@@ -246,3 +187,25 @@ See https://code.visualstudio.com/docs/remote/containers for details.
...
+
+   $ python
+   >>> import pandas
+   >>> print(pandas.__version__)  # note: the exact output may differ
+   2.0.0.dev0+880.g2b9e661fbb.dirty
+
+This will create the new environment, and not touch any of your existing environments,
+nor any existing Python installation.
+
+Note that you will need to repeat this step each time the C extensions change, for example
```
Comment (mroeschke) [inline doc/source/development/contributing_environment.rst:210]: Might be good to call this out in a `.. note::` block

File: doc/source/development/contributing_environment.rst (hunk 1 above)
Comment (MarcoGorelli) [reply doc/source/development/contributing_environment.rst:211]: there's also some in `pandas/io` though

Comment (mroeschke) [reply doc/source/development/contributing_environment.rst]: Oh I didn't know there's some sas cython code there (I think that should be moved to `pandas/_libs` at some point).

Mainly wanted to clarify here what files constitute `C extensions`  (pyx files)

Comment (MarcoGorelli) [reply doc/source/development/contributing_environment.rst]: OK I've gone with your suggestion (it does say "for example", so it's fine if it's not exhaustive, especially if the sas files should be moved to pandas/_libs too)

**Review by mroeschke — APPROVED** [review]
LGTM

**Comment by MarcoGorelli** [comment]
thanks! @natmokval seeing as you went through this not too long ago, reckon it makes it clearer for new contributors?

**Comment by MarcoGorelli** [comment]
@meeseeksdev please backport to 1.5.x

**Comment by MarcoGorelli** [comment]
@meeseeksdev please backport to 1.5.x

**Pull request merged**